src/Core/NotificationHub/NotificationHubPushRegistrationService.cs

```diff
@@ -48,6 +51,12 @@ public async Task CreateOrUpdateRegistrationAsync(string pushToken, string devic
             installation.Tags.Add("deviceIdentifier:" + identifier);
         }
 
+        foreach (var organizationUserDetails in await _organizationUserRepository.GetManyDetailsByUserAsync(
+                     Guid.Parse(userId), OrganizationUserStatusType.Confirmed))
+        {
+            installation.Tags.Add($"organizationId:{organizationUserDetails.OrganizationId}");
+        }
+
```

Comment:
Wouldn't the current context be a better place to grab this info?

Reply:
I did that. In fact it was not working for self-hosted at all. Fixed with 35fab48a

Reply:
and fixed in [d028029](https://github.com/bitwarden/server/pull/4923/commits/d028029270d4e88047e7b5c81f52c2521b0e57df)